packages/plugin-odoo-snippets/src/OdooBindings.ts

```diff
@@ -0,0 +1,163 @@
+    const layout = editor.plugins.get(Layout);
+    const domEngine = layout.engines.dom as DomLayoutEngine;
+
+    function _getVElements(node: Node): VElement[] {
+        const vnodes = domEngine.getNodes(node);
+        for (const vnode of vnodes) {
+            if (!(vnode instanceof VElement)) {
+                throw new Error('VNode is not a VElement');
+            }
+        }
+        return vnodes as VElement[];
+    }
+    const odooCommands = {
+        hasVNode(node: Node): boolean {
+            const vnodes = domEngine.getNodes(node);
+            return !!(vnodes && vnodes.length);
+        },
+        async addClasses(node: Node, classes: string[]): Promise<void> {
+            const params: AddClassParams = {
+                elements: _getVElements(node),
+                classes,
+            };
+            await editor.execCommand<OdooSnippet>('addClasses', params);
+        },
+        async removeClasses(node: Node, classes: string[]): Promise<void> {
+            const params: RemoveClassParams = {
+                elements: _getVElements(node),
+                classes,
+            };
+            await editor.execCommand<OdooSnippet>('removeClasses', params);
+        },
+        async toggleClass(node: Node, klass: string, set?: boolean): Promise<void> {
+            const nodes = domEngine.getNodes(node);
+            if (!nodes) {
+                throw new Error('nodes are empty');
+            }
+            const params: ToggleClassParams = {
+                nodes: nodes,
+                class: klass,
+                set,
+            };
+            await editor.execCommand<OdooSnippet>('toggleClass', params);
+        },
+        async setAttribute(
+            node: Node,
+            attributeName: string,
+            attributeValue: string,
+        ): Promise<void> {
+            const params: SetAttributeParams = {
+                elements: _getVElements(node),
+                attributeName,
+                attributeValue,
+            };
+            await editor.execCommand<OdooSnippet>('setAttribute', params);
+        },
+        async setStyle(
+            node: Node,
+            property: string,
+            value: string,
+            important = false,
+        ): Promise<void> {
+            const params: SetStyleParams = {
+                nodes: domEngine.getNodes(node),
+                property,
+                value,
+                important,
+            };
+            await editor.execCommand<OdooSnippet>('setStyle', params);
+        },
+        // todo: sometimes i need to append. Because the implementation of the insertHTML append
+        // it when the RelativePosition is 'inside', it work. But this is unclear that it will
+        // append.
+        // Change the API to have a clearer distinction and allowing to append/prepend;
+        async insertHtml(rangePoint: HtmlPoint, html: string): Promise<void> {
```

Comment:
I don't understand what a `rangePoint` is. A range is composed of *two* points so there is no such thing as a `rangePoint`. Either it's the `start` or the `end` of the range or it's the `anchor` or the `focus` of the selection.